backup: update kms failed metric via schedule option

kev-cao opened the pull request:
Currently, we update the last successful backup time metric via the `updates_cluster_last_backup_time_metric` _schedule_ option, and the last kms failed metric with the `updates_cluster_monitoring_metric` _backup_ option.

This commit removes the latter option and merges the KMS failure metric into the former.

Epic: None

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/154202)
<!-- Reviewable:end -->

kev-cao committed:
backup: update kms failed metric via schedule option

Currently, we update the last successful backup time metric via the
`updates_cluster_last_backup_time_metric` _schedule_ option, and the
last kms failed metric with the `updates_cluster_monitoring_metric`
_backup_ option.

This commit removes the latter option and merges the KMS failure metric
into the former.

Epic: None

Release note: None
